core/shared/communication/base-communication.ts

```diff
@@ -2,18 +2,30 @@ import { randomUUID } from 'crypto';
 import { Message } from './message';
 import { Operation } from './operation';
 
+type Subscriber = (message: Message) => Promise<Message | undefined>;
```

Comment:
Should we provide just the payload to the subscriber and not the complete message? The subscriber should know the operation, as it specifically subscribed to it. And it's not interested in the id.

Reply:
It would simplify the clients quite a bit, e.g. from
```ts
    this.communication.on(Operation.REQUEST_API, async (message) => {
      const payload = message.payload as RequestApiPayload;
      const response = await _fetch('GET', payload.url, payload.body);
      return { ...message, payload: response };
    });
```
to
```ts
    this.communication.on(Operation.REQUEST_API, (data: RequestApiPayload) => _fetch('GET', data.url, data.body));
```

Reply:
yes good idea will do that